src/main/java/io/gravitee/policy/assignmetrics/AssignMetricsPolicy.java

```diff
@@ -0,0 +1,174 @@
+import io.gravitee.gateway.api.el.EvaluableResponse;
+import io.gravitee.gateway.api.stream.BufferedReadWriteStream;
+import io.gravitee.gateway.api.stream.ReadWriteStream;
+import io.gravitee.gateway.api.stream.SimpleReadWriteStream;
+import io.gravitee.policy.api.PolicyChain;
+import io.gravitee.policy.api.annotations.OnRequest;
+import io.gravitee.policy.api.annotations.OnRequestContent;
+import io.gravitee.policy.api.annotations.OnResponse;
+import io.gravitee.policy.api.annotations.OnResponseContent;
+import io.gravitee.policy.assignmetrics.configuration.AssignMetricsPolicyConfiguration;
+import io.gravitee.policy.assignmetrics.configuration.PolicyScope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Florent CHAMFROY (florent.chamfroy at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class AssignMetricsPolicy {
+
+    private static final Logger logger = LoggerFactory.getLogger(AssignMetricsPolicy.class);
+
+    private static final String REQUEST_VARIABLE = "request";
+    private static final String RESPONSE_VARIABLE = "response";
+
+    private final AssignMetricsPolicyConfiguration assignVariablePolicyConfiguration;
+
+    public AssignMetricsPolicy(final AssignMetricsPolicyConfiguration assignVariablePolicyConfiguration) {
+        this.assignVariablePolicyConfiguration = assignVariablePolicyConfiguration;
+    }
+
+    @OnRequestContent
+    public ReadWriteStream onRequestContent(Request request, ExecutionContext executionContext) {
+        if (
+            assignVariablePolicyConfiguration.getScope() != null &&
+            assignVariablePolicyConfiguration.getScope() == PolicyScope.REQUEST_CONTENT
+        ) {
+            return new BufferedReadWriteStream() {
+                Buffer buffer = Buffer.buffer();
+
+                @Override
+                public SimpleReadWriteStream<Buffer> write(Buffer content) {
+                    buffer.appendBuffer(content);
+                    return this;
+                }
+
+                @Override
+                public void end() {
+                    String content = buffer.toString();
+                    executionContext
+                        .getTemplateEngine()
+                        .getTemplateContext()
+                        .setVariable(REQUEST_VARIABLE, new EvaluableRequest(request, content));
+
+                    // assign
+                    assign(executionContext);
+
+                    if (buffer.length() > 0) {
+                        super.write(buffer);
+                    }
+
+                    super.end();
+                }
+            };
+        }
+
+        return null;
+    }
+
+    @OnResponseContent
+    public ReadWriteStream onResponseContent(Response response, ExecutionContext executionContext) {
+        if (
+            assignVariablePolicyConfiguration.getScope() != null &&
+            assignVariablePolicyConfiguration.getScope() == PolicyScope.RESPONSE_CONTENT
```

Comment:
```suggestion
            PolicyScope.RESPONSE_CONTENT.equals(assignVariablePolicyConfiguration.getScope())
```

OR (as == with enum is null safe)

```suggestion
    assignVariablePolicyConfiguration.getScope() == PolicyScope.RESPONSE_CONTENT
```

Reply:
fix by 951af64